File: node/ts/zkgroup/profiles/ExpiringProfileKeyCredential.ts
//
// Copyright 2022 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray.js';
import * as Native from '../../Native.js';

export default class ExpiringProfileKeyCredential extends ByteArray {
  private readonly __type?: never;

  constructor(contents: Uint8Array) {
    super(contents, Native.ExpiringProfileKeyCredential_CheckValidContents);
  }

  getExpirationTime(): Date {
    return new Date(
      1000 *
        Native.ExpiringProfileKeyCredential_GetExpirationTime(this.contents)
    );
  }
}

File: java/shared/java/org/signal/libsignal/zkgroup/profiles/ExpiringProfileKeyCredential.java
//
// Copyright 2022 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.zkgroup.profiles;

import static org.signal.libsignal.internal.FilterExceptions.filterExceptions;

import java.time.Instant;
import org.signal.libsignal.internal.Native;
import org.signal.libsignal.zkgroup.InvalidInputException;
import org.signal.libsignal.zkgroup.internal.ByteArray;

public final class ExpiringProfileKeyCredential extends ByteArray {
  public ExpiringProfileKeyCredential(byte[] contents) throws InvalidInputException {
    super(contents);
    filterExceptions(
        InvalidInputException.class,
        () -> Native.ExpiringProfileKeyCredential_CheckValidContents(contents));
  }

  public Instant getExpirationTime() {
    return Instant.ofEpochSecond(
        Native.ExpiringProfileKeyCredential_GetExpirationTime(this.contents));
  }
}
